File: javascript/submit_job.mjs
const BASE_URL = process.env.BASE_URL || "https://sandbox.api.neurocad.eu";
const TOKEN = process.env.TOKEN || "replace-me";
const PROJECT_ID = process.env.PROJECT_ID || "prj_demo_001";

async function main() {
  const response = await fetch(`${BASE_URL}/v1/inference/jobs`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${TOKEN}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      project_id: PROJECT_ID,
      task: "field-inference",
      input: {
        material: "steel-a36",
        resolution: "medium",
      },
      labels: ["public-js-example"],
    }),
  });

  console.log(JSON.stringify(await response.json(), null, 2));
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: python/submit_job.py
"""Reference example for submitting a NeuroCAD job from Python."""

from __future__ import annotations

import json
from urllib import request


BASE_URL = "https://sandbox.api.neurocad.eu"
TOKEN = "replace-me"


def submit_solver_export(project_id: str) -> dict:
    payload = {
        "project_id": project_id,
        "task": "solver-export",
        "input": {
            "material": "aluminum-6061",
            "export_format": "solver-bundle",
            "quality_profile": "balanced",
        },
        "labels": ["public-reference"],
    }

    data = json.dumps(payload).encode("utf-8")
    headers = {
        "Authorization": f"Bearer {TOKEN}",
        "Content-Type": "application/json",
    }
    req = request.Request(
        f"{BASE_URL}/v1/inference/jobs",
        data=data,
        headers=headers,
        method="POST",
    )
    with request.urlopen(req) as response:  # noqa: S310 - example only
        return json.loads(response.read().decode("utf-8"))


if __name__ == "__main__":
    print("Reference example. Replace TOKEN before use.")
    result = submit_solver_export("prj_demo_001")
    print(json.dumps(result, indent=2))
